Content.Client/Explosion/ExplosionOverlay.cs

```diff
@@ -48,22 +33,20 @@ protected override void Draw(in OverlayDrawArgs args)
 
         var xforms = _entMan.GetEntityQuery<TransformComponent>();
 
-        if (ActiveExplosion != null && ActiveExplosion.Map == args.Viewport.Eye?.Position.MapId)
+        foreach (var (comp, appearance) in _entMan.EntityQuery<ExplosionVisualsComponent, AppearanceComponent>(true))
         {
-            DrawExplosion(drawHandle, args.WorldBounds, ActiveExplosion, Index, xforms);
-        }
+            if (!appearance.TryGetData(ExplosionAppearanceData.Progress, out int index))
+                return;
```

Comment:
Should this be continue?

Reply:
I guess, though if nothing else breaks it shouldn't ever get triggered anyways.